cigaradvisor/scripts/templates/blog-post.js

```diff
@@ -88,8 +89,36 @@ export default async function decorate(main) {
   div.append(articleHeaderBlockEl);
   main.prepend(div);
 
-  const section = document.createElement('div');
-  section.append(buildBlock('article-navigation', []));
-  main.append(section);
+  const articleList = document.createElement('div');
+  const ul = document.createElement('ul');
+  const posts = await loadPosts();
+  // 2 posts from the same author “most recent by published date”
+  const authorPosts = posts.filter((post) => authorLink.includes(post.author) && post.path !== window.location.pathname);
+  if (authorPosts) {
+    if (authorPosts[0]) ul.innerHTML = `<li><a href="${authorPosts[0].path}">${authorPosts[0].path}</a></li>`;
+    if (authorPosts[1]) ul.innerHTML += `<li><a href="${authorPosts[1].path}">${authorPosts[1].path}</a></li>`;
+  }
+  // 1 post from the same category “most recent by published date”
+  const categoryPosts = posts.filter((post) => category.includes(post.category) && !authorPosts.includes(post) && post.path !== window.location.pathname);
+  if (categoryPosts && categoryPosts[0]) {
+    ul.innerHTML += `<li><a href="${categoryPosts[0].path}">${categoryPosts[0].path}</a></li>`;
+  }
+  // 1 random post
+  const randomPost = posts.filter((post) => !authorPosts.includes(post) && !categoryPosts.includes(post) && post.path !== window.location.pathname);
```

Comment:
This is supposed to be 1 random post from the list on the _Must Reads_ page. not one random from anywhere.